[*] Start mutation process:
   - targets: graph2.py
   - tests: ts-3-5/test_3-5_1-0_319.py
[*] 6 tests passed:
   - test_3-5_1-0_319 [0.19848 s]
[*] Start mutants generation and execution:
   - [#   1] CDI graph2: 
--------------------------------------------------------------------------------
  22:     adjacent(n1, n2): returns True if there is an edge connecting n1 and n2,
  23:     False if not, raises an error if either of the supplied nodes are not in g.
  24:     """
  25:     
- 26:     def __init__(self, data=None):
- 27:         '''Initialize graph.'''
- 28:         self.graph = {}
- 29:         if data:
- 30:             for i in data:
- 31:                 self.add_node(i)
- 32:     
+ 26:     @classmethod
+ 27:     def __init__(self, data=None):
+ 28:         '''Initialize graph.'''
+ 29:         self.graph = {}
+ 30:         if data:
+ 31:             for i in data:
+ 32:                 self.add_node(i)
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.17461 s] survived
   - [#   2] CDI graph2: 
--------------------------------------------------------------------------------
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
- 33:     def nodes(self):
- 34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
- 36:     
+ 33:     @classmethod
+ 34:     def nodes(self):
+ 35:         '''Return a list of all nodes in the graph.'''
+ 36:         return list(self.graph.keys())
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
--------------------------------------------------------------------------------
[0.22480 s] killed by ts-3-5/test_3-5_1-0_319.py::test_initial_empty_nodes

graph_instance = <graph2.Graph object at 0x78fff7147910>

    def test_initial_empty_nodes(graph_instance):
>       assert graph_instance.nodes() == []

ts-3-5/test_3-5_1-0_319.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:36: AttributeError
   - [#   3] CDI graph2: 
--------------------------------------------------------------------------------
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
- 37:     def edges(self):
- 38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
- 40:     
+ 37:     @classmethod
+ 38:     def edges(self):
+ 39:         '''Return a list of all edges in the graph.'''
+ 40:         return [edge for edges in self.graph.values() for edge in edges]
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
--------------------------------------------------------------------------------
[0.18461 s] survived
   - [#   4] CDI graph2: 
--------------------------------------------------------------------------------
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
- 41:     def add_node(self, n):
- 42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
- 44:     
+ 41:     @classmethod
+ 42:     def add_node(self, n):
+ 43:         '''Add a new node to the graph.'''
+ 44:         self.graph.setdefault(n, set())
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
--------------------------------------------------------------------------------
[0.20352 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_node

graph_instance = <graph2.Graph object at 0x78fff719a470>

    def test_add_node(graph_instance):
>       graph_instance.add_node('A')

ts-3-5/test_3-5_1-0_319.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 'A'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:44: AttributeError
   - [#   5] CDI graph2: 
--------------------------------------------------------------------------------
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
- 45:     def add_edge(self, n1, n2):
- 46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
- 50:     
+ 45:     @classmethod
+ 46:     def add_edge(self, n1, n2):
+ 47:         '''Add new edge to the graph.'''
+ 48:         self.graph.setdefault(n1, set([n2]))
+ 49:         self.graph.setdefault(n2, set())
+ 50:         self.graph[n1].add(n2)
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
--------------------------------------------------------------------------------
[0.23018 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_edge

graph_instance = <graph2.Graph object at 0x78fff71fb400>

    def test_add_edge(graph_instance):
        graph_instance.add_node('C')
        graph_instance.add_node('D')
>       graph_instance.add_edge('C', 'D')

ts-3-5/test_3-5_1-0_319.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n1 = 'C', n2 = 'D'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:48: AttributeError
   - [#   6] CDI graph2: 
--------------------------------------------------------------------------------
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
- 51:     def del_node(self, n):
- 52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
- 54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
- 56:     
+ 51:     @classmethod
+ 52:     def del_node(self, n):
+ 53:         """Delete the node 'n' from the graph."""
+ 54:         del self.graph[n]
+ 55:         for k in self.graph:
+ 56:             self.graph[k].discard(n)
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.21153 s] killed by ts-3-5/test_3-5_1-0_319.py::test_del_node

graph_instance = <graph2.Graph object at 0x78fff719b580>

    def test_del_node(graph_instance):
        graph_instance.add_node('E')
>       graph_instance.del_node('E')

ts-3-5/test_3-5_1-0_319.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 'E'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:54: AttributeError
   - [#   7] CDI graph2: 
--------------------------------------------------------------------------------
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
- 57:     def del_edge(self, n1, n2):
- 58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
- 60:     
+ 57:     @classmethod
+ 58:     def del_edge(self, n1, n2):
+ 59:         '''Delete the edge connecting n1 and n2.'''
+ 60:         self.graph[n1].remove(n2)
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
--------------------------------------------------------------------------------
[0.21253 s] killed by ts-3-5/test_3-5_1-0_319.py::test_del_edge

graph_instance = <graph2.Graph object at 0x78fff7114ee0>

    def test_del_edge(graph_instance):
        graph_instance.add_node('F')
        graph_instance.add_node('G')
        graph_instance.add_edge('F', 'G')
>       graph_instance.del_edge('F', 'G')

ts-3-5/test_3-5_1-0_319.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n1 = 'F', n2 = 'G'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:60: AttributeError
   - [#   8] CDI graph2: 
--------------------------------------------------------------------------------
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
- 61:     def has_node(self, n):
- 62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
- 64:     
+ 61:     @classmethod
+ 62:     def has_node(self, n):
+ 63:         """Return boolean if 'n' is in the graph."""
+ 64:         return n in self.graph
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
--------------------------------------------------------------------------------
[0.18152 s] survived
   - [#   9] CDI graph2: 
--------------------------------------------------------------------------------
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
- 65:     def neighbors(self, n):
- 66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
- 68:     
+ 65:     @classmethod
+ 66:     def neighbors(self, n):
+ 67:         '''Return the list of all nodes connected to n by edges.'''
+ 68:         return self.graph[n]
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21206 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_edge

graph_instance = <graph2.Graph object at 0x78fff73799c0>

    def test_add_edge(graph_instance):
        graph_instance.add_node('C')
        graph_instance.add_node('D')
        graph_instance.add_edge('C', 'D')
>       assert 'D' in graph_instance.neighbors('C')

ts-3-5/test_3-5_1-0_319.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 'C'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:68: AttributeError
   - [#  10] CDI graph2: 
--------------------------------------------------------------------------------
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
- 69:     def adjacent(self, n1, n2):
- 70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 69:     @classmethod
+ 70:     def adjacent(self, n1, n2):
+ 71:         '''Return boolean if there is an edge connecting n1 and n2.'''
+ 72:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.18138 s] survived
   - [#  11] COI graph2: 
--------------------------------------------------------------------------------
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
  28:         self.graph = {}
- 29:         if data:
+ 29:         if not data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
--------------------------------------------------------------------------------
[0.20504 s] incompetent

TypeError: graph2:30: TypeError: 'NoneType' object is not iterable

   - [#  12] COI graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         return n not in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.18627 s] survived
   - [#  13] COI graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         return n2 not in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.19170 s] survived
   - [#  14] OIL graph2: 
--------------------------------------------------------------------------------
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
- 32:     
+ 32:                 break
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.16988 s] survived
   - [#  15] OIL graph2: 
--------------------------------------------------------------------------------
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
- 56:     
+ 56:             break
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.18907 s] survived
   - [#  16] RIL graph2: 
--------------------------------------------------------------------------------
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
- 30:             for i in data:
+ 30:             for i in reversed(data):
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
--------------------------------------------------------------------------------
[0.19016 s] survived
   - [#  17] RIL graph2: 
--------------------------------------------------------------------------------
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
- 54:         for k in self.graph:
+ 54:         for k in reversed(self.graph):
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
--------------------------------------------------------------------------------
[0.18668 s] survived
   - [#  18] SDI graph2: 
--------------------------------------------------------------------------------
  22:     adjacent(n1, n2): returns True if there is an edge connecting n1 and n2,
  23:     False if not, raises an error if either of the supplied nodes are not in g.
  24:     """
  25:     
- 26:     def __init__(self, data=None):
- 27:         '''Initialize graph.'''
- 28:         self.graph = {}
- 29:         if data:
- 30:             for i in data:
- 31:                 self.add_node(i)
- 32:     
+ 26:     @staticmethod
+ 27:     def __init__(self, data=None):
+ 28:         '''Initialize graph.'''
+ 29:         self.graph = {}
+ 30:         if data:
+ 31:             for i in data:
+ 32:                 self.add_node(i)
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.21095 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_319.py:13: TypeError: Graph.__init__() missing 1 required positional argument: 'self'

   - [#  19] SDI graph2: 
--------------------------------------------------------------------------------
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
- 33:     def nodes(self):
- 34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
- 36:     
+ 33:     @staticmethod
+ 34:     def nodes(self):
+ 35:         '''Return a list of all nodes in the graph.'''
+ 36:         return list(self.graph.keys())
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
--------------------------------------------------------------------------------
[0.21837 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_319.py:16: TypeError: Graph.nodes() missing 1 required positional argument: 'self'

   - [#  20] SDI graph2: 
--------------------------------------------------------------------------------
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
- 37:     def edges(self):
- 38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
- 40:     
+ 37:     @staticmethod
+ 38:     def edges(self):
+ 39:         '''Return a list of all edges in the graph.'''
+ 40:         return [edge for edges in self.graph.values() for edge in edges]
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
--------------------------------------------------------------------------------
[0.18635 s] survived
   - [#  21] SDI graph2: 
--------------------------------------------------------------------------------
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
- 41:     def add_node(self, n):
- 42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
- 44:     
+ 41:     @staticmethod
+ 42:     def add_node(self, n):
+ 43:         '''Add a new node to the graph.'''
+ 44:         self.graph.setdefault(n, set())
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
--------------------------------------------------------------------------------
[0.20846 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_319.py:19: TypeError: Graph.add_node() missing 1 required positional argument: 'n'

   - [#  22] SDI graph2: 
--------------------------------------------------------------------------------
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
- 45:     def add_edge(self, n1, n2):
- 46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
- 50:     
+ 45:     @staticmethod
+ 46:     def add_edge(self, n1, n2):
+ 47:         '''Add new edge to the graph.'''
+ 48:         self.graph.setdefault(n1, set([n2]))
+ 49:         self.graph.setdefault(n2, set())
+ 50:         self.graph[n1].add(n2)
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
--------------------------------------------------------------------------------
[0.22409 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_319.py:30: TypeError: Graph.add_edge() missing 1 required positional argument: 'n2'

   - [#  23] SDI graph2: 
--------------------------------------------------------------------------------
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
- 51:     def del_node(self, n):
- 52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
- 54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
- 56:     
+ 51:     @staticmethod
+ 52:     def del_node(self, n):
+ 53:         """Delete the node 'n' from the graph."""
+ 54:         del self.graph[n]
+ 55:         for k in self.graph:
+ 56:             self.graph[k].discard(n)
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.20997 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_319.py:35: TypeError: Graph.del_node() missing 1 required positional argument: 'n'

   - [#  24] SDI graph2: 
--------------------------------------------------------------------------------
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
- 57:     def del_edge(self, n1, n2):
- 58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
- 60:     
+ 57:     @staticmethod
+ 58:     def del_edge(self, n1, n2):
+ 59:         '''Delete the edge connecting n1 and n2.'''
+ 60:         self.graph[n1].remove(n2)
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
--------------------------------------------------------------------------------
[0.21304 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_319.py:42: TypeError: Graph.del_edge() missing 1 required positional argument: 'n2'

   - [#  25] SDI graph2: 
--------------------------------------------------------------------------------
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
- 61:     def has_node(self, n):
- 62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
- 64:     
+ 61:     @staticmethod
+ 62:     def has_node(self, n):
+ 63:         """Return boolean if 'n' is in the graph."""
+ 64:         return n in self.graph
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
--------------------------------------------------------------------------------
[0.19329 s] survived
   - [#  26] SDI graph2: 
--------------------------------------------------------------------------------
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
- 65:     def neighbors(self, n):
- 66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
- 68:     
+ 65:     @staticmethod
+ 66:     def neighbors(self, n):
+ 67:         '''Return the list of all nodes connected to n by edges.'''
+ 68:         return self.graph[n]
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21136 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_319.py:31: TypeError: Graph.neighbors() missing 1 required positional argument: 'n'

   - [#  27] SDI graph2: 
--------------------------------------------------------------------------------
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
- 69:     def adjacent(self, n1, n2):
- 70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 69:     @staticmethod
+ 70:     def adjacent(self, n1, n2):
+ 71:         '''Return boolean if there is an edge connecting n1 and n2.'''
+ 72:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.18849 s] survived
   - [#  28] SDL graph2: 
--------------------------------------------------------------------------------
  24:     """
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
- 28:         self.graph = {}
+ 28:         pass
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
--------------------------------------------------------------------------------
[0.21143 s] killed by ts-3-5/test_3-5_1-0_319.py::test_initial_empty_nodes

graph_instance = <graph2.Graph object at 0x78fff7194430>

    def test_initial_empty_nodes(graph_instance):
>       assert graph_instance.nodes() == []

ts-3-5/test_3-5_1-0_319.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff7194430>

>   ???
E   AttributeError: 'Graph' object has no attribute 'graph'

graph2:35: AttributeError
   - [#  29] SDL graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 pass
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.19915 s] survived
   - [#  30] SDL graph2: 
--------------------------------------------------------------------------------
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
+ 35:         pass
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
--------------------------------------------------------------------------------
[0.21118 s] killed by ts-3-5/test_3-5_1-0_319.py::test_initial_empty_nodes

graph_instance = <graph2.Graph object at 0x78fff729a2c0>

    def test_initial_empty_nodes(graph_instance):
>       assert graph_instance.nodes() == []
E       AssertionError

ts-3-5/test_3-5_1-0_319.py:16: AssertionError
   - [#  31] SDL graph2: 
--------------------------------------------------------------------------------
  35:         return list(self.graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
+ 39:         pass
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
--------------------------------------------------------------------------------
[0.19212 s] survived
   - [#  32] SDL graph2: 
--------------------------------------------------------------------------------
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
+ 43:         pass
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
--------------------------------------------------------------------------------
[0.21263 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_node

graph_instance = <graph2.Graph object at 0x78fff7145cf0>

    def test_add_node(graph_instance):
        graph_instance.add_node('A')
>       assert 'A' in graph_instance.nodes()
E       AssertionError

ts-3-5/test_3-5_1-0_319.py:20: AssertionError
   - [#  33] SDL graph2: 
--------------------------------------------------------------------------------
  43:         self.graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
+ 47:         pass
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
--------------------------------------------------------------------------------
[0.18464 s] survived
   - [#  34] SDL graph2: 
--------------------------------------------------------------------------------
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
+ 48:         pass
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
--------------------------------------------------------------------------------
[0.18787 s] survived
   - [#  35] SDL graph2: 
--------------------------------------------------------------------------------
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
+ 49:         pass
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
--------------------------------------------------------------------------------
[0.21454 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_edge

graph_instance = <graph2.Graph object at 0x78fff6f616c0>

    def test_add_edge(graph_instance):
        graph_instance.add_node('C')
        graph_instance.add_node('D')
        graph_instance.add_edge('C', 'D')
>       assert 'D' in graph_instance.neighbors('C')
E       AssertionError

ts-3-5/test_3-5_1-0_319.py:31: AssertionError
   - [#  36] SDL graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             pass
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.18765 s] survived
   - [#  37] SDL graph2: 
--------------------------------------------------------------------------------
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
+ 59:         pass
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
--------------------------------------------------------------------------------
[0.21462 s] killed by ts-3-5/test_3-5_1-0_319.py::test_del_edge

graph_instance = <graph2.Graph object at 0x78fff6fb5de0>

    def test_del_edge(graph_instance):
        graph_instance.add_node('F')
        graph_instance.add_node('G')
        graph_instance.add_edge('F', 'G')
        graph_instance.del_edge('F', 'G')
>       assert 'G' not in graph_instance.neighbors('F')
E       AssertionError

ts-3-5/test_3-5_1-0_319.py:43: AssertionError
   - [#  38] SDL graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         pass
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.19342 s] survived
   - [#  39] SDL graph2: 
--------------------------------------------------------------------------------
  63:         return n in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
+ 67:         pass
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21357 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_319.py:31: TypeError: argument of type 'NoneType' is not iterable

   - [#  40] SDL graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         pass
--------------------------------------------------------------------------------
[0.18100 s] survived
   - [#  41] SVD graph2: 
--------------------------------------------------------------------------------
  24:     """
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
- 28:         self.graph = {}
+ 28:         graph = {}
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [#  42] SVD graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.18723 s] survived
   - [#  43] SVD graph2: 
--------------------------------------------------------------------------------
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
+ 35:         return list(graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
--------------------------------------------------------------------------------
[0.22774 s] killed by ts-3-5/test_3-5_1-0_319.py::test_initial_empty_nodes

graph_instance = <graph2.Graph object at 0x78fff6f19a20>

    def test_initial_empty_nodes(graph_instance):
>       assert graph_instance.nodes() == []

ts-3-5/test_3-5_1-0_319.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff6f19a20>

>   ???
E   NameError: name 'graph' is not defined

graph2:35: NameError
   - [#  44] SVD graph2: 
--------------------------------------------------------------------------------
  35:         return list(self.graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
+ 39:         return [edge for edges in graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
--------------------------------------------------------------------------------
[0.18620 s] survived
   - [#  45] SVD graph2: 
--------------------------------------------------------------------------------
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
+ 43:         graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
--------------------------------------------------------------------------------
[0.20876 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_node

graph_instance = <graph2.Graph object at 0x78fff73c2200>

    def test_add_node(graph_instance):
>       graph_instance.add_node('A')

ts-3-5/test_3-5_1-0_319.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff73c2200>, n = 'A'

>   ???
E   NameError: name 'graph' is not defined

graph2:43: NameError
   - [#  46] SVD graph2: 
--------------------------------------------------------------------------------
  43:         self.graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
+ 47:         graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
--------------------------------------------------------------------------------
[0.21652 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_edge

graph_instance = <graph2.Graph object at 0x78fff6f3f1c0>

    def test_add_edge(graph_instance):
        graph_instance.add_node('C')
        graph_instance.add_node('D')
>       graph_instance.add_edge('C', 'D')

ts-3-5/test_3-5_1-0_319.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff6f3f1c0>, n1 = 'C', n2 = 'D'

>   ???
E   NameError: name 'graph' is not defined

graph2:47: NameError
   - [#  47] SVD graph2: 
--------------------------------------------------------------------------------
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
+ 48:         graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
--------------------------------------------------------------------------------
[0.20923 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_edge

graph_instance = <graph2.Graph object at 0x78fff6f19c90>

    def test_add_edge(graph_instance):
        graph_instance.add_node('C')
        graph_instance.add_node('D')
>       graph_instance.add_edge('C', 'D')

ts-3-5/test_3-5_1-0_319.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff6f19c90>, n1 = 'C', n2 = 'D'

>   ???
E   NameError: name 'graph' is not defined

graph2:48: NameError
   - [#  48] SVD graph2: 
--------------------------------------------------------------------------------
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
+ 49:         graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
--------------------------------------------------------------------------------
[0.21195 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_edge

graph_instance = <graph2.Graph object at 0x78fff715c550>

    def test_add_edge(graph_instance):
        graph_instance.add_node('C')
        graph_instance.add_node('D')
>       graph_instance.add_edge('C', 'D')

ts-3-5/test_3-5_1-0_319.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff715c550>, n1 = 'C', n2 = 'D'

>   ???
E   NameError: name 'graph' is not defined

graph2:49: NameError
   - [#  49] SVD graph2: 
--------------------------------------------------------------------------------
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
+ 53:         del graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
--------------------------------------------------------------------------------
[0.21635 s] killed by ts-3-5/test_3-5_1-0_319.py::test_del_node

graph_instance = <graph2.Graph object at 0x78fff7116a70>

    def test_del_node(graph_instance):
        graph_instance.add_node('E')
>       graph_instance.del_node('E')

ts-3-5/test_3-5_1-0_319.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff7116a70>, n = 'E'

>   ???
E   NameError: name 'graph' is not defined

graph2:53: NameError
   - [#  50] SVD graph2: 
--------------------------------------------------------------------------------
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
- 54:         for k in self.graph:
+ 54:         for k in graph:
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
--------------------------------------------------------------------------------
[0.20305 s] killed by ts-3-5/test_3-5_1-0_319.py::test_del_node

graph_instance = <graph2.Graph object at 0x78fff6f3e380>

    def test_del_node(graph_instance):
        graph_instance.add_node('E')
>       graph_instance.del_node('E')

ts-3-5/test_3-5_1-0_319.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff6f3e380>, n = 'E'

>   ???
E   NameError: name 'graph' is not defined

graph2:54: NameError
   - [#  51] SVD graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.19222 s] survived
   - [#  52] SVD graph2: 
--------------------------------------------------------------------------------
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
+ 59:         graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
--------------------------------------------------------------------------------
[0.21803 s] killed by ts-3-5/test_3-5_1-0_319.py::test_del_edge

graph_instance = <graph2.Graph object at 0x78fff715db10>

    def test_del_edge(graph_instance):
        graph_instance.add_node('F')
        graph_instance.add_node('G')
        graph_instance.add_edge('F', 'G')
>       graph_instance.del_edge('F', 'G')

ts-3-5/test_3-5_1-0_319.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff715db10>, n1 = 'F', n2 = 'G'

>   ???
E   NameError: name 'graph' is not defined

graph2:59: NameError
   - [#  53] SVD graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         return n in graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.19528 s] survived
   - [#  54] SVD graph2: 
--------------------------------------------------------------------------------
  63:         return n in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
+ 67:         return graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21883 s] killed by ts-3-5/test_3-5_1-0_319.py::test_add_edge

graph_instance = <graph2.Graph object at 0x78fff6f3e170>

    def test_add_edge(graph_instance):
        graph_instance.add_node('C')
        graph_instance.add_node('D')
        graph_instance.add_edge('C', 'D')
>       assert 'D' in graph_instance.neighbors('C')

ts-3-5/test_3-5_1-0_319.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x78fff6f3e170>, n = 'C'

>   ???
E   NameError: name 'graph' is not defined

graph2:67: NameError
   - [#  55] SVD graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         return n2 in neighbors(n1)
--------------------------------------------------------------------------------
[0.18714 s] survived
   - [#  56] ZIL graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 break
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.18253 s] survived
   - [#  57] ZIL graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             break
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.18305 s] survived
[*] Mutation score [13.16181 s]: 42.6%
   - all: 57
   - killed: 20 (35.1%)
   - survived: 27 (47.4%)
   - incompetent: 10 (17.5%)
   - timeout: 0 (0.0%)
